lib/AST/Decl.cpp

```diff
@@ -1031,7 +1031,7 @@ unsigned PatternBindingDecl::getPatternEntryIndexForVarDecl(const VarDecl *VD) c
     ++Result;
   }
   
-  assert(0 && "PatternBindingDecl doesn't bind the specified VarDecl!");
+  llvm_unreachable("PatternBindingDecl doesn't bind the specified VarDecl!");
```

Comment:
Should we add some sort of `swift_warn_unreachable()` that allows recovery in production builds? Then we could replace all uses of `assert(0)` with either `llvm_unreachable()` or `swift_warn_unreachable()`.